### PR TITLE
feat(router): added merchant custom name support for payment link

### PR DESCRIPTION
**Pull request by sahkal**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

Add merchant custom name support for payment link

### Additional Changes

- [x] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables
`"payment_link_object": {
        "link_expiry" : "2025-10-09T16:23:10Z",
        "merchant_custom_domain_name": null,
        "custom_merchant_name": "sahkal"
    }`

## How did you test it?

Tested locally

Screenshot 2023-10-25 at 4 51 19 PM
Screenshot 2023-10-25 at 4 51 28 PM

## Testing Curl

`curl --location 'http://localhost:8080/payments' \
--header 'Content-Type: application/json' \
--header 'Accept: application/json' \
--header 'api-key: dev_bKPPICtnI7RGJVIg97fO3Vzejk87kpO2Cggy9MULP3FcXxeQkFBHdHS6buRmk0ty' \
--data '{
    "amount": 3423,
    "currency": "USD",
    "confirm": false,
    "customer_id": "sahkal",
    "return_url": "https://google.com",
    "description": "For selling Tea",
    "order_details": [
        {
            "product_name": "Tea",
            "quantity": 1,
            "amount": 10,
            "product_img_link": null
        },
        {
            "product_name": "Tea",
            "quantity": 1,
            "amount": 10,
            "product_img_link": "https://thumbs.dreamstime.com/b/indian-tea-spices-masala-chai-33827904.jpg"
        },
        {
            "product_name": "Tea",
            "quantity": 1,
            "amount": 10,
            "product_img_link": "https://thumbs.dreamstime.com/b/indian-tea-spices-masala-chai-33827904.jpg"
        },
        {
            "product_name": "Tea",
            "quantity": 1,
            "amount": 10,
            "product_img_link": "https://thumbs.dreamstime.com/b/indian-tea-spices-masala-chai-33827904.jpg"
        },
        {
            "product_name": "Tea",
            "quantity": 1,
            "amount": 10,
            "product_img_link": "https://thumbs.dreamstime.com/b/indian-tea-spices-masala-chai-33827904.jpg"
        }
    ],
    "payment_link_object": {
        "link_expiry": "2025-10-09T16:23:10Z",
        "merchant_custom_domain_name": null,
        "custom_merchant_name": "sahkal"
    }
}'`

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

**Comment by Narayanbhat166** [comment]
Will this override the merchant name? If not passed in payment link object will this field be taken from merchant account?

File: crates/router/src/core/payment_link.rs [hunk 1]
```diff
@@ -55,20 +56,14 @@ pub async fn intiate_payment_link_flow(
...
-        .async_and_then(|pli| async move {
-            db.find_payment_link_by_payment_link_id(pli)
-                .await
-                .ok()?
-                .fulfilment_time
-                .ok_or(errors::ApiErrorResponse::PaymentNotFound)
-                .ok()
-        })
+        .async_and_then(
+            |pli| async move { db.find_payment_link_by_payment_link_id(pli).await.ok() },
```
Comment (Narayanbhat166) [inline crates/router/src/core/payment_link.rs:63]: why is `ok()` called here? raise the error. Do not ignore such errors

File: crates/api_models/src/payments.rs [hunk 2]
```diff
@@ -3138,11 +3140,11 @@ pub struct PaymentLinkDetails {
...
     pub client_secret: String,
     pub payment_id: String,
-    #[serde(with = "common_utils::custom_serde::iso8601")]
-    pub expiry: PrimitiveDateTime,
+    #[serde(with = "common_utils::custom_serde::iso8601::option")]
+    pub expiry: Option<PrimitiveDateTime>,
     pub merchant_logo: String,
     pub return_url: String,
-    pub merchant_name: crypto::OptionalEncryptableName,
+    pub merchant_name: String,
```
Comment (Narayanbhat166) [inline crates/api_models/src/payments.rs:3147]: this should be encryptable field, when serialized it will be exposed

**Review by jarnura — DISMISSED** [review]
LGTM

**Review by Narayanbhat166 — APPROVED** [review]

**Review by jarnura — APPROVED** [review]

**Pull request merged**